.github/workflows/github-actions-essential-ci.yml

```diff
@@ -344,3 +344,53 @@ jobs:
       - name: clean up
         run: ./build/github/cleanup-engflow-keys.sh
         if: always()
+  cockroach-microbench-ci:
+    runs-on: [ self-hosted, basic_runner_group ]
+    timeout-minutes: 60
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 1  # Fetch only the latest commit
+
+      - name: Get the latest commit of the target branch
+        id: get_latest_commit_target_branch
+        run: |
+          # Get the target branch name
+          TARGET_BRANCH=${{ github.event.pull_request.base.ref }}
+          
+          # Fetch the latest commit of the target branch
+          git fetch origin $TARGET_BRANCH --depth=1
+          
+          # Get the latest commit hash of the target branch
+          LATEST_COMMIT=$(git rev-parse FETCH_HEAD)
+          
+          # Output the latest commit hash
+          echo "Latest commit on target branch ($TARGET_BRANCH): $LATEST_COMMIT"
+          
+          # Set the latest commit hash as an output variable
+          echo "::set-output name=latest_commit::$LATEST_COMMIT"
+      - uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.sha || github.ref }}
+      - name: compute metadata
+        run: echo GITHUB_ACTIONS_BRANCH=${{ github.event.pull_request.number || github.ref_name}} >> "$GITHUB_ENV"
+      - run: ./build/github/get-engflow-keys.sh
+      - name: run scripts
+        run: ./build/github/cockroach-microbench-ci.sh
+        env:
+          COMPARE_THRESHOLD: "5.00"
+          PUSH_STEP: ${{ github.event_name == 'push' }}
+          BASE_SHA: ${{ steps.get_latest_commit_target_branch.outputs.latest_commit }}
+
+          # TODO(sambhav-jain-16): add Performance note check and use this flag. Currently set to false (https://github.com/cockroachdb/cockroach/issues/106661)
+          SKIP_COMPARISON: false
+      - run: ./build/github/prepare-summarize-build.sh
```

Comment:
`prepare-summarize-build.sh` must come before the `run scripts` step and after the `get-engflow-keys.sh` step; please check out the other builds in this file for reference.

Reply:
Oops, Thanks for pointing out !